Fitness/Views/GymFacilitiesView.swift

```diff
@@ -0,0 +1,483 @@
+        default:
+            return
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        guard let unwrappedFacilityName = self.facilityName else {
+            return 0
+        }
+        
+        switch unwrappedFacilityName {
+        case .helenNewman:
+            if tableView == basketballTableView! {
+                return facilityData[0].isDropped ? 7 : 0
+            } else if tableView == bowlingTableView! {
+                return facilityData[1].isDropped ? 7 : 0
+            } else {
+                return facilityData[2].isDropped ? 7 : 0
+            }
+        case .teagle:
+            return facilityData[0].isDropped ? 7 : 0
+        default:
+            return 0
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let unwrappedFacilityName = self.facilityName else {
+            return UITableViewCell()
+        }
+        
+        switch unwrappedFacilityName {
+        case .helenNewman:
+            let cell = tableView.dequeueReusableCell(withIdentifier: FacilityHoursCell.identifier, for: indexPath) as! FacilityHoursCell
+            
+            if tableView == basketballTableView {
+                cell.hoursLabel.text = facilityData[0].data[indexPath.row].0
+                cell.dayLabel.text = facilityData[0].data[indexPath.row].1
+                cell.textLabel
+                
+                return cell
+            } else if tableView == bowlingTableView {
+                cell.hoursLabel.text = facilityData[1].data[indexPath.row].0
+                cell.dayLabel.text = facilityData[1].data[indexPath.row].1
+                
+                return cell
+            } else {
+                cell.hoursLabel.text = facilityData[2].data[indexPath.row].0
+                cell.dayLabel.text = facilityData[2].data[indexPath.row].1
+                
+                return cell
+            }
+        case .teagle:
+            let cell = tableView.dequeueReusableCell(withIdentifier: FacilityHoursCell.identifier, for: indexPath) as! FacilityHoursCell
+            cell.hoursLabel.text = facilityData[0].data[indexPath.row].0
+            cell.dayLabel.text = facilityData[0].data[indexPath.row].1
+            
+            return cell
+        default:
+            return UITableViewCell()
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        guard let unwrappedFacilityName = self.facilityName else {
+            return nil
+        }
+        
+        switch unwrappedFacilityName {
+        case .helenNewman:
+            let header = tableView.dequeueReusableHeaderFooterView(withIdentifier: FacilityHoursHeaderView.identifier) as! FacilityHoursHeaderView
+            var gesture: FacilitiesGestureRecognizer!
+            
+            if tableView == basketballTableView {
```

Comment:
Switching on the table view seems like an odd way to do this, is there a reason this way was chosen over another?